// @ionic-native/serial/index.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import { Observable } from 'rxjs';
var SerialOriginal = /** @class */ (function (_super) {
    __extends(SerialOriginal, _super);
    function SerialOriginal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    SerialOriginal.prototype.requestPermission = function (options) { return cordova(this, "requestPermission", { "successIndex": 1, "errorIndex": 2 }, arguments); };
    SerialOriginal.prototype.open = function (options) { return cordova(this, "open", {}, arguments); };
    SerialOriginal.prototype.write = function (data) { return cordova(this, "write", {}, arguments); };
    SerialOriginal.prototype.writeHex = function (data) { return cordova(this, "writeHex", {}, arguments); };
    SerialOriginal.prototype.read = function () { return cordova(this, "read", {}, arguments); };
    SerialOriginal.prototype.registerReadCallback = function () { return cordova(this, "registerReadCallback", { "observable": true }, arguments); };
    SerialOriginal.prototype.close = function () { return cordova(this, "close", {}, arguments); };
    SerialOriginal.pluginName = "Serial";
    SerialOriginal.plugin = "cordovarduino";
    SerialOriginal.pluginRef = "serial";
    SerialOriginal.repo = "https://github.com/xseignard/cordovarduino";
    SerialOriginal.platforms = ["Android", "Ubuntu"];
    return SerialOriginal;
}(IonicNativePlugin));
var Serial = new SerialOriginal();
export { Serial };
